agent/src/util/device_plugin_service.rs

```diff
@@ -335,19 +382,39 @@ impl InstanceDevicePlugin {
             );
             let mut akri_annotations = HashMap::new();
             let mut akri_device_properties = HashMap::new();
+            let mut akri_devices = HashMap::<String, Device>::new();
             for device_usage_id in request.devices_i_ds {
                 trace!(
                     "InstanceDevicePlugin::allocate - for Instance {} processing request for device usage slot id {}",
                     &dps.instance_name,
                     device_usage_id
                 );
 
+                if let Err(e) = try_update_instance_device_usage(
```

Comment:
i can't imagine how changing where try_update_instance_device_usage is called relative to the other code in allocate would matter ... but am curious why the move?

Reply:
nothing special, since we return error when try_update_instance_device_usage fails, call it earlier then we don't need to waste time to add annotations, and other bookkeeping works.